src/test/java/com/musala/atmosphere/websocket/util/CustomDataDeserializationTest.java

```diff
@@ -0,0 +1,155 @@
+import com.musala.atmosphere.commons.RoutingAction;
+import com.musala.atmosphere.commons.beans.DeviceAcceleration;
+import com.musala.atmosphere.commons.geometry.Bounds;
+import com.musala.atmosphere.commons.geometry.Point;
+import com.musala.atmosphere.commons.ui.tree.AccessibilityElement;
+import com.musala.atmosphere.commons.util.Pair;
+import com.musala.atmosphere.commons.util.structure.tree.Node;
+import com.musala.atmosphere.commons.util.structure.tree.Tree;
+import com.musala.atmosphere.commons.websocket.message.MessageAction;
+import com.musala.atmosphere.commons.websocket.message.RequestMessage;
+import com.musala.atmosphere.commons.websocket.message.ResponseMessage;
+
+/**
+ * Tests the deserialization of the {@link ResponseMessage} custom response data
+ *
+ * @author dimcho.nedev
+ *
+ */
+public class CustomDataDeserializationTest extends BaseDataSerialization {
+
+    @Test
+    public void deviceInformationResponseDataDeserializationTest() {
+        DeviceInformation deviceInformation = createDeviceInfoObject();
+
+        DeviceInformation deserializedResponseData = (DeviceInformation) getExpectedResponseData(RoutingAction.GET_DEVICE_INFORMATION,
+                                                                                                 deviceInformation);
+
+        Assert.assertEquals(TestConst.DEVICE_API_LEVEL, deserializedResponseData.getApiLevel());
+        Assert.assertEquals(new Integer(TestConst.RESOLUTION_X), deserializedResponseData.getResolution().getKey());
+        Assert.assertEquals(new Integer(TestConst.RESOLUTION_Y), deserializedResponseData.getResolution().getValue());
+        Assert.assertEquals(new Pair<Integer, Integer>(TestConst.RESOLUTION_X, TestConst.RESOLUTION_Y),
+                            deserializedResponseData.getResolution());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void treeOfAccessibilityElementsResponseDataDeserializationTest() {
+        // setup elements
+        AccessibilityElement rootElement = new AccessibilityElement();
+        AccessibilityElement firstChildElement = new AccessibilityElement();
+        AccessibilityElement secondChildElement = new AccessibilityElement();
+        AccessibilityElement childOfFirstChildElement = new AccessibilityElement();
+        Bounds expectedBounds = new Bounds(new Point(0, 0), new Point(4, 4));
+
+        rootElement.setPath(TestConst.TEST_PATH);
+        firstChildElement.setText("first_child_text");
+        secondChildElement.setEnabled(true);
+        childOfFirstChildElement.setBounds(expectedBounds);
+
+        // setup the nodes
+        Node<AccessibilityElement> firstChildNode = new Node<AccessibilityElement>(firstChildElement);
+        Node<AccessibilityElement> secondChildNode = new Node<AccessibilityElement>(secondChildElement);
+        Node<AccessibilityElement> childOfsecondChildNode = new Node<AccessibilityElement>(childOfFirstChildElement);
```

Comment:
here the name does not match the actual purpose
Furthermore, please capitalise Second